packages/server/billing/helpers/teamLimitsCheck.ts

```diff
@@ -114,16 +114,27 @@ export const maybeRemoveRestrictions = async (orgId: string, dataLoader: DataLoa
   }
 
   if (!(await isLimitExceeded(orgId, dataLoader))) {
-    await r
-      .table('Organization')
-      .get(orgId)
-      .update({
-        tierLimitExceededAt: null,
-        scheduledLockAt: null,
-        lockedAt: null,
-        updatedAt: new Date()
-      })
-      .run()
+    const billingLeaders = await getBillingLeaders(orgId, dataLoader)
+    const billingLeadersIds = billingLeaders.map((billingLeader) => billingLeader.id)
+    await Promise.all([
+      r
+        .table('Organization')
+        .get(orgId)
+        .update({
+          tierLimitExceededAt: null,
+          scheduledLockAt: null,
+          lockedAt: null,
+          updatedAt: new Date()
+        })
+        .run(),
+      r
+        .table('OrganizationUser')
+        .getAll(r.args(billingLeadersIds), {index: 'userId'})
+        .filter({orgId})
+        .update({suggestedTier: 'starter'})
```

Comment:
-1 Because you're loading the billingLeaders via dataloader, we need to update or reset them in the cache.